src/Consolonia.Core/Drawing/PixelBufferImplementation/PixelBuffer.cs

```diff
@@ -68,6 +70,25 @@ public void Set<TUserObject>(PixelBufferCoordinate point, Func<Pixel, TUserObjec
             this[point] = changeAction(this[point], userObject);
         }
 
+        /// <summary>
+        ///     Clears old pixel caret position and sets new caret position
+        /// </summary>
+        /// <param name="point"></param>
+        public void SetCaretPosition(PixelBufferCoordinate point)
+        {
+            // clear old caret position by merging in IsCaret = false
+            Pixel oldCaretPixel = _buffer[_caretPosition.X, _caretPosition.Y];
```

Comment:
The reason I did this is there was a check and exception thrown on the second caret found. I assumed this meant your intent was for only a single caret. 

I can simply remove the check... But what is the scenario for two carets? Can that happen in avalonia?

Reply:
So I tried not toggling and removing the check and what happens is that it completely broke cursor navigation on pages with multiple input controls. 

I think way I have it is correct, or at least if there is a scenario where we need this it can be a future PR.

Reply:
> I can simply remove the check... But what is the scenario for two carets? Can that happen in avalonia?

Opposite, I'm saying that we should throw an exception (may be optional tho) when two carets have been tried to show.
it should not be legit to try to show two cursors, we can not determine which one is supposed to be correct.
Any control is supposed to show the caret only when focused. Avalonia tracks the focus: any focus change starts with LostFocus in old place and only then GotFocus in new place.

Reply:
> So I tried not toggling and removing the check and what happens is that it completely broke cursor navigation on pages with multiple input controls.

How to reproduce it?

> I think way I have it is correct, or at least if there is a scenario where we need this it can be a future PR.
Yeah, we can just extract this to a separate topic. But as I understand it worked before and stopped with this PR, or?

Reply:
@tomlm we can move this discussion to a separate topic to merge this PR if you think would be better

Reply:
It confuses me only because I expected there are no attempts to show double cursor which we just can not do anyway. If this attempt happens, I would better investigate why and fix that instead of allowing this.

Reply:
got it, I will dig into why this is happening now. It must be something about my changes.

Reply:
> got it, I will dig into why this is happening now. It must be something about my changes.

If you push intermediate changes with the repro, I could try to investigate in parallel.

Reply:
I found it, you are right. 
I introduced a bug in Pixel.Blend().  I was merging IsCaret when merging regular foreground pixels by falling through to the the isCaret section instead of returning the blended pixel. I have fixed it, and the exception doesn't throw anymore. 

Good catch!